qiskit/algorithms/optimizers/steppable_optimizer.py

```diff
@@ -0,0 +1,304 @@
+
+        """
+        raise NotImplementedError
+
+    def _callback_wrapper(self) -> None:
+        """
+        Wraps the callback function to accomodate each optimizer.
+        """
+        pass
+
+    def step(self) -> None:
+        """
+        Performs one step in the optimization process.
+        This method composes :meth:`~.ask`, :meth:`~.evaluate`, and :meth:`~.tell` to make a "step"
+        in the optimization process.
+        """
+        ask_data = self.ask()
+        tell_data = self.evaluate(ask_data=ask_data)
+        self.tell(ask_data=ask_data, tell_data=tell_data)
+
+    @abstractmethod
+    def start(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> None:
+        """Populates the state of the optimizer with the data provided and sets all the counters to 0.
+
+        Args:
+            fun: Function to minimize.
+            x0: Initial point.
+            jac: Function to compute the gradient.
+            bounds: Bounds of the search space.
+
+        """
+        raise NotImplementedError
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+        """Minimizes the function.
+        For well behaved functions the user can call this method to minimize a function.
+        If the user wants more control on how to evaluate the function a custom loop can be
+        created using :meth:`~.ask` and :meth:`~.tell` and evaluating the function manually.
+
+        Args:
+            fun: Function to minimize.
+            x0: Initial point.
+            jac: Function to compute the gradient.
+            bounds: Bounds of the search space.
+
+        Returns:
+            Object containing the result of the optimization.
+
+        """
+        self.start(x0=x0, fun=fun, jac=jac, bounds=bounds)
+        while self.continue_condition():
+            self.step()
+            self._callback_wrapper()
+        return self.create_result()
+
+    @abstractmethod
+    def create_result(self) -> OptimizerResult:
+        """
+        Returns the result of the optimization.
+        All the information needed to create such a result should be stored in the optimizer state
+        and will typically contain the best point found, the function value and gradient at that point,
+        the number of function and gradient evaluation and the number of iterations in the optimization.
+
+        Returns:
+            The result of the optimization process.
+
+        """
+        raise NotImplementedError
+
+    def continue_condition(self) -> bool:
+        """Condition that indicates the optimization process should come to an end.
```

Comment:
I guess this has a bearing on my earlier comment around continuation/termination. Arguably the docstring here is talking about a termination condition (inverse of continue) not a continue condition. `Condition that indicates the optimization process should continue or not` would seem to me to be how to think about it, which is how the returns reads too

Reply:
Yes, the documentation is missleading. I changed it.